sswf-core/src/main/scala/com/bazaarvoice/sswf/service/util.scala

```diff
@@ -7,6 +7,6 @@ object util {
   // fingerprint a certain step configuration so we can be sure we are executing the step we intend to.
   // So instead of a real version number, we use the timeout. If that changes, the step is logically different.
   private[service] def stepToVersion[StepEnum <: (Enum[StepEnum] with WorkflowStep)](step: StepEnum): String =
-    s"${step.timeoutSeconds}" + ".hbtm" /*feature: consolidate timeouts to just heartbeat*/
+    s"${step.versionOverride}.${step.timeoutSeconds}"
```

Comment:
you're concerned that a running workflow may be unable to complete if you upgrade in the middle?

I think this won't happen, but I'll write a quick test.